packages/react-storage/src/components/StorageBrowser/views/LocationActionView/CopyView/CopyViewProvider.tsx

```diff
@@ -89,15 +87,6 @@ export function CopyViewProvider({
       }
     : getActionCompleteMessage({ counts: statusCounts });
 
-  const foldersMessage = !hasFoldersInitialized
-    ? undefined
-    : getListFoldersResultsMessage({
-        hasError: hasFoldersError,
-        message: foldersErrorMessage,
-        folders: pageItems,
-        query,
-      });
```

Comment:
Conditionally returning `undefined` here was the root cause of the memo issue, moved the call to `getListFoldersResultsMessage` itself inside `FoldersMessageControl` and removed the init flag